packages/oauth/src/install-provider.ts

```diff
@@ -245,32 +285,84 @@ export class InstallProvider {
    * The return value is an Array of Promises made up of the resolution of each token refresh attempt.
    */
   private async refreshExpiringTokens(tokensToRefresh: string[]): Promise<OAuthV2TokenRefreshResponse[]> {
-    const client = new WebClient(undefined, this.clientOptions);
-
-    const refreshPromises = tokensToRefresh.map(async (refreshToken) => await client.oauth.v2.access({
-      client_id: this.clientId,
-      client_secret: this.clientSecret,
-      grant_type: 'refresh_token',
-      refresh_token: refreshToken,
-    }).catch((e) => e) as OAuthV2TokenRefreshResponse);
```

Comment:
I don't think this `as OAuthV2TokenRefreshResponse` is correct.

Reply:
`catch((e) => e)` is not necessary

Reply:
No, it is still necessary but the code can be improved more. updated in the latest revision

Reply:
Confirmed the new implementation works better for error patterns